qiskit_nature/drivers/molecule.py

```diff
@@ -62,7 +64,15 @@ def __init__(
         """
         Molecule._check_consistency(geometry, masses)
 
+        # if unit is Bohr, convert geometry to unit angstrom
+        if unit == UnitsType.BOHR:
+            n_geometry: List[Tuple[str, List[float]]] = []
+            for atom, xyz in geometry:
+                ang_xyz = [e * BOHR for e in xyz]
+                n_geometry.append((atom, ang_xyz))
+            geometry = n_geometry
```

Comment:
As mentioned in my previous comment we should refrain from doing this conversion inside of `Molecule`. I do not believe that any of the code here in this class explicitly relies on the units being `Angstrom`.
Instead, when a driver gets passed a `Molecule`, it should deal with whatever unit the `Molecule` is in, by itself.

We could think about providing a small utility method on the class, e.g. `Molecule.angstrom_to_bohr` and vice versa for convenience. But I am not sure that it is really necessary for a computation as simple as this.

Reply:
And related to this would be the `__str__`  method where I think if you print the Molecule you would expect to see the same values you created it with. Maybe the method could/should display the units too in that case for completeness.

Reply:
Ah yes I just picked the first occurrence to add this comment. Thanks for pointing this out :+1: